Sources/Fluent/Entity/Entity.swift

```diff
@@ -1,9 +1,51 @@
+public final class Storage {
+    public init() {}
+
+    fileprivate var exists: Bool = false
+    fileprivate var id: Node? = nil
+}
+
+public protocol Storable: class {
+    var storage: Storage { get }
+}
+
+extension Storable {
+    /// Whether or not entity was retrieved from database.
+    ///
+    /// This value shouldn't be interacted w/ external users
+    /// w/o explicit knowledge.
+    ///
+    /// General implementation should just be `let storage = Storage()`
+    public internal(set) var exists: Bool {
+        get {
+            return storage.exists
+        }
+        set {
+            storage.exists = newValue
+        }
+    }
+
+    public var id: Node? {
+        get {
+            return storage.id
+        }
+        set {
+            // TODO: Should we log errs if storage.id already has value? 
```

Comment:
Should that delete the old user? Either way, sounds like onus is on end user, I'll remove the comment 👍